### PR TITLE
feat: upgrade workspace to Angular 17

### PR DESCRIPTION
**Pull request by arturovt**

### CODE REVIEW

**Comment by markwhitfeld** [comment]
@arturovt I see that there is quite a bit of reformatting of files in this PR.
Could you split that reformatting work into a separate PR to make this one easier to review?

**Comment by arturovt** [comment]
I just ran Nx migration and then when committing changes it ran pre-commit hook and formatted files...

**Comment by arturovt** [comment]
@markwhitfeld I have reverted formatting changes.

File: integration/app/app.component.ts [hunk 1]
```diff
@@ -39,12 +38,16 @@ export class AppComponent implements OnInit {
...
 
   greeting: string;
-  todos$: Observable<Todo[]> = this._store.select(TodoState);
-  pandas$: Observable<Todo[]> = this._store.select(TodoState.getPandas);
-  pizza$: Observable<Pizza> = this._store.select(TodosState.getPizza);
-  injected$: Observable<string> = this._store.select(TodosState.getInjected);
+  todos: Signal<Todo[]> = this._store.selectSignal(TodoState);
+  pandas: Signal<Todo[]> = this._store.selectSignal(TodoState.getPandas);
+  pizza: Signal<Pizza> = this._store.selectSignal(TodosState.getPizza);
+  injected: Signal<string> = this._store.selectSignal(TodosState.getInjected);
```
Comment (markwhitfeld) [inline integration/app/app.component.ts:44]: I think that we should have a signal version, and an observable version of this component. WDYT?

File: integration/app/app.config.ts [hunk 2]
```diff
@@ -42,7 +41,7 @@ export const appConfig: ApplicationConfig = {
       [TodosState, TodoState],
       { developmentMode: !environment.production, selectorOptions: {} },
       withNgxsFormPlugin(),
-      withNgxsLoggerPlugin({ logger: console, collapsed: false, disabled: true }),
+      withNgxsLoggerPlugin({ logger: console, collapsed: false, disabled: false }),
```
Comment (markwhitfeld) [inline integration/app/app.config.ts:44]: Do we want to enable logging as part of this PR?
Or was this just left after some debugging?

File: integration/jest.config.js [hunk 3]
```diff
@@ -12,16 +12,18 @@ module.exports = {
...
   moduleNameMapper,
   coverageDirectory: '../coverage/integration',
   transform: {
-    '^.+\\.(ts|js|html)$': 'jest-preset-angular'
+    '^.+\\.(ts|mjs|js|html)$': [
+      'jest-preset-angular',
+      {
+        isolatedModules: true,
+        tsconfig: '<rootDir>/tsconfig.spec.json',
+        stringifyContentPathRegex: '\\.(html|svg)$'
```
Comment (markwhitfeld) [inline integration/jest.config.js:24]: Why was this needed?
If it was auto-generated by NX, then it's just their implementation details, but if it was something that was added by us for NGXS, then is this something we should add as guidance in our docs?

File: package.json [hunk 4]
```diff
@@ -79,7 +79,7 @@
     "build:integration:ssr": "yarn build:integration && nx server integration --configuration production",
     "// - TESTING": "Test all modules #requires yarn build:packages",
     "test:integration": "nx run integration:test --colors --run-in-band",
-    "test:types": "dtslint packages/store/types --localTs node_modules/typescript/lib",
+    "test:types": "cd packages/store/types && yarn test:types",
```
Comment (markwhitfeld) [inline package.json:82]: Something to do with this change seems to be failing the tests.
I see that there is a `package.json` file with a `yarn.lock`but no install is run from ci for this folder.
- Potentially we could add a `workspaces` property in the main `package.json` that includes this folder to make that happen, but I'm not sure how `yarn` does it, compared to modern `npm`.
- Or we could modify the `.github/actions/setup/action.yml` file to also run the install from that folder.
- Or we could add a `yarn install... ` command into the types folder's `package.json` to do the install just in time. Just like how we do for the integration apps.

File: .github/workflows/trunk.yml [hunk 5]
```diff
@@ -78,7 +78,7 @@ jobs:
     strategy:
       matrix:
         script:
-          - 'integration:ng15'
+          - 'integration:ng16'
```
Comment (markwhitfeld) [inline .github/workflows/trunk.yml:81]: Should we add `- 'integration:ng17'` here too?
```suggestion
          - 'integration:ng16'
          - 'integration:ng17'
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -78,7 +78,7 @@ jobs:
     strategy:
       matrix:
         script:
-          - 'integration:ng15'
+          - 'integration:ng16'
```
Comment (markwhitfeld) [inline .github/workflows/release.yml:81]: Should we add `- 'integration:ng17'` here too?
```suggestion
          - 'integration:ng16'
          - 'integration:ng17'
```

File: .github/workflows/pr-validation.yml [hunk 7]
```diff
@@ -106,7 +105,7 @@ jobs:
         run: yarn ${{ matrix.script }}
 
       - name: Upload an integration test artifact
-        if: ${{ matrix.script == 'integration:ng15' }}
+        if: ${{ matrix.script == 'integration:ng16' }}
```
Comment (markwhitfeld) [inline .github/workflows/pr-validation.yml:108]: This should run for all integration tests.
```suggestion
        if: ${{ startsWith( matrix.script, 'integration:' ) }}
```

File: .github/actions/download-integration-test-artifacts/action.yml [hunk 8]
```diff
@@ -18,8 +18,8 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - name: Download hello-world-ng15 artifacts
+    - name: Download hello-world-ng16 artifacts
       uses: actions/download-artifact@v3
       with:
-        name: hello-world-ng15
+        name: hello-world-ng16
         path: ${{ inputs.path }}
```
Comment (markwhitfeld) [inline .github/actions/download-integration-test-artifacts/action.yml:25]: This step should be repeated for ng 17 too.

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -96,7 +96,7 @@ jobs:
         run: yarn ${{ matrix.script }}
 
       - name: Upload an integration test artifact
-        if: ${{ matrix.script == 'integration:ng15' }}
+        if: ${{ matrix.script == 'integration:ng16' }}
```
Comment (markwhitfeld) [inline .github/workflows/release.yml:99]: This should run for all integration tests.
```suggestion
        if: ${{ startsWith( matrix.script, 'integration:' ) }}
```

File: .github/workflows/trunk.yml [hunk 10]
```diff
@@ -96,7 +96,7 @@ jobs:
         run: yarn ${{ matrix.script }}
 
       - name: Upload an integration test artifact
-        if: ${{ matrix.script == 'integration:ng15' }}
+        if: ${{ matrix.script == 'integration:ng16' }}
```
Comment (markwhitfeld) [inline .github/workflows/trunk.yml:99]: This should run for all integration tests.
```suggestion
        if: ${{ startsWith( matrix.script, 'integration:' ) }}
```

**Review by markwhitfeld — CHANGES_REQUESTED** [review]
There is so much going on in this PR that it was super difficult to review.
I have a number of suggested changes and questions.
If it is easy to split this into smaller PRs then that would be awesome. If not, then let's press ahead to try to get this PR in.

File: integration/jest.config.js (hunk 3 above)
Comment (markwhitfeld) [reply integration/jest.config.js:24]: Made by NX

**Review by markwhitfeld — COMMENTED** [review]
Approved in principle, but we should not merge this until we are building towards the next release that drops support for ng15 and earlier.

**Review by markwhitfeld — APPROVED** [review]
I did a deeper review, and it doesn't look like this will affect our supported Angular versions.

**Pull request merged**